MetaMask/metamask-extension
# chore: update Trezor Connect to v9.4.0, remove workarounds

This PR includes changes from PR #26749 by @martykan.


## **Description**

> With MV3, MetaMask started to use Trezor Connect inside an offscreen environment, in a way that was previously unsupported and required a workaround by patching the Trezor Connect library.
> 
> In recent versions of Trezor Connect, the library can handle working in an offscreen environment correctly, without the need for the workaround, which could cause issues with compatibility.
> 
> This PR removes the patch and updates the Trezor Connect library to the latest version (v9.4.0).
> The change in manifest.json is due to new URL parameters, Firefox needs the asterisk at the end to match the URL with them.
> I haven't removed the WebUSB device request which was added on MetaMask's side in relation to the workaround, since it can improve UX of the pairing process, but it could be removed if desired.
> 
> The dependency update affects Lavamoat, I am including the policy changes in my commit, however let me know if you would like me to remove them and handle them using your own process.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27112?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

> 1. Open accounts dropdown, "Add hardware wallet"
> 2. Select Trezor
> 3. Follow prompts to connect the device
> 4. See a list of accounts from the Trezor

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**

- [x ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x ] I've completed the PR template to the best of my ability

## **Pre-merge reviewer checklist**
